simulator/virtual_machine.go

```diff
@@ -400,7 +400,36 @@ func (vm *VirtualMachine) applyExtraConfig(spec *types.VirtualMachineConfigSpec)
 		val := c.GetOptionValue()
 		key := strings.TrimPrefix(extraConfigKey(val.Key), "SET.")
 		if key == val.Key {
-			vm.Config.ExtraConfig = append(vm.Config.ExtraConfig, c)
+			keyIndex := -1
+			for i := range vm.Config.ExtraConfig {
+				bov := vm.Config.ExtraConfig[i]
+				if bov == nil {
+					continue
+				}
+				ov := bov.GetOptionValue()
+				if ov == nil {
+					continue
+				}
+				if ov.Key == key {
+					keyIndex = i
+					break
+				}
+			}
+			if keyIndex < 0 {
+				vm.Config.ExtraConfig = append(vm.Config.ExtraConfig, c)
+			} else {
+				if s, ok := val.Value.(string); ok && s == "" {
```

Comment:
Curious, is an empty value always treated as removing the key?  Is there a case where someone genuinely wants to pass in an empty string?  Official docs on the `OptionValue` type are not that helpful: https://vdc-repo.vmware.com/vmwb-repository/dcr-public/1ef6c336-7bef-477d-b9bb-caa1767d7e30/82521f49-9d9a-42b7-b19b-9e6cd9b30db1/vim.option.OptionValue.html

Reply:
See: https://vdc-download.vmware.com/vmwb-repository/dcr-public/c476b64b-c93c-4b21-9d76-be14da0148f9/04ca12ad-59b9-4e1c-8232-fd3d4276e52c/SDK/vsphere-ws/docs/ReferenceGuide/vim.vm.ConfigSpec.html
> If the key is already present, it will be reset with the new value provided. Otherwise, a new option is added. Keys with empty values will be removed.

Reply:
In that way I suppose I should check to see if the value is not just a string, but an empty value of any type?